contracts/wrappers/algebra-v1.2/PositionManagerAbstractAlgebraV1_2.sol

```diff
@@ -0,0 +1,442 @@
+
+/**
+ * @title PositionManagerAbstractAlgebra
+ * @dev Extension of PositionManagerAbstract for managing Algebra V1.2 positions with added features like custom token swapping.
+ */
+abstract contract PositionManagerAbstractAlgebraV1_2 is
+  PositionManagerAlgebraBase
+{
+  /**
+   * @notice Creates a new position wrapper and initializes it with specified liquidity.
+   * @param _dustReceiver Address to receive any leftover tokens after transactions.
+   * @param _token0 Address of the first token in the liquidity pair.
+   * @param _token1 Address of the second token in the liquidity pair.
+   * @param _name Name for the new wrapper token.
+   * @param _symbol Symbol for the new wrapper token.
+   * @param params Parameters for initializing the liquidity position.
+   * @return Address of the newly created position wrapper.
+   */
+  function createNewWrapperPositionAndDeposit(
+    address _dustReceiver,
+    address _token0,
+    address _token1,
+    string memory _name,
+    string memory _symbol,
+    WrapperFunctionParameters.PositionMintParamsAlgebra memory params
+  ) external notPaused nonReentrant returns (address) {
+    if (_dustReceiver == address(0)) revert ErrorLibrary.InvalidAddress();
+
+    // Create and initialize a new wrapper position
+    IPositionWrapper positionWrapper = createNewWrapperPosition(
+      _token0,
+      _token1,
+      _name,
+      _symbol,
+      params._tickLower,
+      params._tickUpper
+    );
+
+    // Initialize the Uniswap V3 position with specified liquidity and mint wrapper tokens
+    _initializePositionAndDeposit(_dustReceiver, positionWrapper, params);
+
+    // Return the address of the new wrapper position
+    return address(positionWrapper);
+  }
+
+  /**
+   * @notice Initializes a new Uniswap V3 position with liquidity for the first time and mints wrapper tokens.
+   * @notice Adjusts the price range and liquidity of an existing Algebra V3 position.
+   * @param _positionWrapper The wrapper of the position to be adjusted.
+   * @param params The liquidity parameters including the desired amounts of token0 and token1, and slippage protections.
+   */
+  function initializePositionAndDeposit(
+    address _dustReceiver,
+    IPositionWrapper _positionWrapper,
+    WrapperFunctionParameters.InitialMintParamsAlgebra memory params
+  ) external notPaused nonReentrant {
+    // Mint the new Algebra V3 position using the provided liquidity parameters.
+    _initializePositionAndDeposit(
+      _dustReceiver,
+      _positionWrapper,
+      WrapperFunctionParameters.PositionMintParamsAlgebra({
+        _amount0Desired: params._amount0Desired,
+        _amount1Desired: params._amount1Desired,
+        _amount0Min: params._amount0Min,
+        _amount1Min: params._amount1Min,
+        _tickLower: _positionWrapper.initialTickLower(),
+        _tickUpper: _positionWrapper.initialTickUpper(),
+        _deployer: params._deployer
```

Comment:
This deployer, I guess means, the person who is deploying the pool/nft, so this will be always msg.sedner?

Reply:
No, the deployer has to be passed for to the external position when minting a new position. It's zero address for now but might change depending on the pool